src/tests/sqlsmith/src/validation.rs

```diff
@@ -15,8 +15,9 @@
 //! Provides validation logic for expected errors.
 use risingwave_expr::ExprError;
 
-fn is_division_by_zero_err(db_error: &str) -> bool {
-    db_error.contains(&ExprError::DivisionByZero.to_string())
+/// Ignore errors related to `0`.
+fn is_zero_err(db_error: &str) -> bool {
+    db_error.contains(&ExprError::DivisionByZero.to_string()) || db_error.contains("can't be zero")
```

Comment:
Guess soon we will need to ignore the following errors as well:
```
test=# select pow(0, -1);
ERROR:  zero raised to a negative power is undefined
test=# select pow(-1, 0.5);
ERROR:  a negative number raised to a non-integer power yields a complex result
```

Right now our behavior is inconsistent with PG, with the former returning `Numeric out of range` (already handled) and latter evaluates to `NaN` (no error).

This suggest we may need to categorize certain errors as "safe to ignore in sqlsmith" proactively. But I cannot see clearly where the boundary is. Or maybe the current approach of adding new ones reactively is good enough? Just raising the discussion and would like to hear what your thoughts.

Reply:
Some preliminary thoughts:
There are several criteria which I consider when ignoring errors:
1. Is it error caused by user? e.g. `DivisionByZeroError`
1. Is the error message expected?
1. Is it complex to workaround it?

If yes to all 3, can consider including it in ignored errors.

I think the currently set of errors ignored by sqlsmith includes most if not all the categories which are safe to ignore.

What is done reactively is including errors in these existing categories.
We can add more errors proactively to these categories, I suspect several errors are not ignored:
- zero errors
- out of range errors
- negative errors

Criteria 3 is particularly important for sqlsmith. My thinking is that it is worth the trade-off: 
1. Pro: Keep sqlsmith simple, easy to maintain.
2. Tradeoff: Perhaps some errors should not be returned, but these are rare.

I think there's still some room for improvements which are lower effort but higher reward: see #3896 .

Eventually we can do something like:
- Generate only non erroring arguments for functions.
- Workaround unsupported features (`nested_loop_join`, `subquery_unnesting`).

Reply:
actually they occur more frequently than expected. We should generate queries which do not contain these errors. #7866